crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs

```diff
@@ -119,7 +122,53 @@ fn to_iceberg_predicate(expr: &Expr) -> TransformedResult {
                 _ => TransformedResult::NotTransformed,
             }
         }
-        Expr::Cast(c) => to_iceberg_predicate(&c.expr),
+        Expr::Cast(c) => {
+            if DataType::Date32 == c.data_type || DataType::Date64 == c.data_type {
+                match c.expr.as_ref() {
+                    Expr::Literal(ScalarValue::Utf8(Some(literal))) => {
+                        let date = literal.split('T').next();
```

Comment:
I would be reluctant to do any string manipulation here.

Reply:
Instead, I would add support for the datetime format in `date_from_str`:

https://github.com/apache/iceberg-rust/blob/75c7e8d67942fc6e29413d8011c508bdba95abe9/crates/iceberg/src/spec/values.rs#L1459-L1466